src/CaptureClient/include/CaptureClient/ApiEventProcessor.h

```diff
@@ -40,6 +42,10 @@ class ApiEventProcessor {
   void ProcessApiTrackUint(const orbit_grpc_protos::ApiTrackUint& grpc_api_track_uint);
   void ProcessApiTrackUint64(const orbit_grpc_protos::ApiTrackUint64& grpc_api_track_uint64);
 
+  void SetApiEventIdProvider(std::unique_ptr<ApiEventIdProvider> provider) {
```

Comment:
From what I understand, if this is not called we just crash. Should we just pass it to the constructor?

Reply:
Unfortunately, its not that easy. This relies on the CaptureOptions (for the function ids), which we only know when we receive the capture started event (at least in the case of capture loading). The alternative to move the constructor call up the chain and only construct the ApiEventIdProvider OnCaptureStarted. But this also only shifts the issue...

Reply:
But at least that deserves a comment somewhere.

Reply:
Ah right, so the idea would be to call this method when the `CaptureStarted` message is received?